### PR TITLE
add Stalmarck plugin to CI

### PR DESCRIPTION
**Pull request by palmskog**
[Stalmarck](https://github.com/coq-community/stalmarck) is a verified tautology checker, formerly a Coq Contrib. It consists of a Coq library and a plugin with a tactic that uses code extracted from the library. The project is maintained by me as part of Coq-Community.

The CI script first builds and installs the library. Then, Dune extracts code and builds and installs the plugin. I tried to follow other Dune projects in CI like MathComp Word.

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by palmskog** [comment]
@Zimmi48 since this is a Coq-Community project, could you maybe review when opportunity arises? The build seems to work.

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by Zimmi48 — APPROVED** [review]

**Review by ejgallego — APPROVED** [review]

**Comment by SkySkimmer** [comment]
Tested locally it said
~~~
+ dune build @install
Entering directory '/home/gaetan/dev/coq/coq'
Error: Don't know about directory _build_ci/stalmarck specified on the
command line!
make: *** [Makefile.ci:135: ci-stalmarck] Error 1
~~~

**Comment by palmskog** [comment]
If `dune build @install` doesn't work for some reason (it works fine for me locally outside Coq's CI), then I have no idea how it can be made to work, since for example `dune build @install -p coq-stalmarck-tactic` will not due work due to this Dune issue: https://github.com/ocaml/dune/issues/4814

@ejgallego any advice here? There's no way I can change from Dune to something else, and both opam package definitions (which trigger Dune issue) need to live in the root directory.

File: dev/ci/ci-stalmarck.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+git_download stalmarck
+
+if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
+
+( cd "${CI_BUILD_DIR}/stalmarck"
+  make
+  make install
+
+  dune build @install
```
Comment (SkySkimmer) [inline dev/ci/ci-stalmarck.sh:16]: Going by the mathcomp_word script maybe
```suggestion
  dune build @install -p coq-stalmarck-tactic
```
would work?

Comment (palmskog) [reply dev/ci/ci-stalmarck.sh:16]: No, this never worked for me locally outside Coq's CI. This triggers Dune issue https://github.com/ocaml/dune/issues/4814
```shell
Error: The following <package>.install are missing:
- _build/default/coq-stalmarck.install
```

Comment (palmskog) [reply dev/ci/ci-stalmarck.sh:16]: @SkySkimmer could you try the following locally? It's the only hack workaround I can think that doesn't require me to purge useful files from the repo:

```shell
  make
  make install

  rm coq-stalmarck.opam
  dune build @install -p coq-stalmarck-tactic
  dune install -p coq-stalmarck-tactic --prefix="$CI_INSTALL_DIR"
```

Comment (SkySkimmer) [reply dev/ci/ci-stalmarck.sh:16]: That seems to work
NB `rm -f` is better so that the script can be rerun without failing

Comment (palmskog) [reply dev/ci/ci-stalmarck.sh]: Done in 2728389bb21, do you need to do the "full CI" thing again?

Comment (SkySkimmer) [reply dev/ci/ci-stalmarck.sh]: I had the magic label on ;)

**Comment by ejgallego** [comment]
The first error is because dune will ignore directories starting with `_` by default, this `_build_ci` is ignored.

Actually not sure how the solutions you propose are working, are they building stuff?

That opam thing was always a PITA in Dune.

I guess a more principled approach would be to uncomment `(dirs :standard _build_ci)` from Coq for the CI run, and quality the build alias `dune build ./@install` or `dune build _build_ci/stalkmark/@install`

**Comment by palmskog** [comment]
As shown in [this job](https://gitlab.com/coq/coq/-/jobs/3334410847) stalmarck now builds and installs fine with the hack. The idea is simply to remove the "other" opam package definition from the checked out code, so the Dune bug isn't triggered.

File: dev/ci/ci-stalmarck.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+git_download stalmarck
+
+if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
+
+# work around https://github.com/ocaml/dune/issues/4814
+( cd "${CI_BUILD_DIR}/stalmarck"
+  make
+  make install
+
+  rm -f coq-stalmarck.opam
```
Comment (SkySkimmer) [inline dev/ci/ci-stalmarck.sh:17]:
```suggestion
( cd "${CI_BUILD_DIR}/stalmarck"
  make
  make install

  rm -f coq-stalmarck.opam # work around https://github.com/ocaml/dune/issues/4814
```

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**